fronts-client/src/actions/Cards.ts

```diff
@@ -214,7 +214,44 @@ const updateCardMetaWithPersist = (persistTo: PersistTo) =>
  * Group ids remain consistent, even if the group is hidden (when maxItems is set to 0), so we can use the id to determine the group.
  */
 
-const mayLowerCardBoostLevelForDestinationGroup = (
+const boostLevels = ['default', 'boost', 'megaboost', 'gigaboost'] as const;
+
+const minimumGroupBoostLevel = (groupId: number) => {
+	// boost is the smallest boost level for `Big`
+	if (groupId === 1) return 'boost';
+	// megaboost is the smallest boost level for `Very Big`
+	if (groupId === 2) return 'megaboost';
+	return 'default';
+};
+
+/**
+ * If the card is moved to a splash (group 3) group, we always set the boost level to default
+ * If the card already has the minimum boost level for the group it is moving to,
+ * we don't want to go any lower.
+ * Otherwise, we decrease the boost level by 1.
+ */
+const getBoostLevel = (groupId: number, boostIndex: number) => {
```

Comment:
Maybe saying it's the current index is a tad clearer? 

```suggestion
const getBoostLevel = (groupId: number, currentBoostIndex: number) => {
```